meta/src/store/state_machine.rs

```diff
@@ -647,6 +647,14 @@ impl StateMachine {
     }
 
     fn process_add_date_node(&self, cluster: &str, node: &NodeInfo) -> CommandResp {
+        for (_, value) in
+            children_data::<NodeInfo>(&KeyPath::data_nodes(cluster), self.db.clone()).iter()
+        {
+            // Check whether the ip of the same cluster has been added
+            if value.http_addr == node.http_addr || value.tcp_addr == node.tcp_addr {
```

Comment:
node_id should also be checked.

Reply:
If you check the node_id, will you output an error when the node_id is duplicated?